### PR TITLE
Fix #13 Created materials listing page

### LINKED ISSUE #13: Create a page for listing materials
We need a page for viewing all the materials corresponding to any of the categories.
The user upon interacting with any of the category cards in the main landing page will be redirected to this page.

1. First is the navbar.
2. Second, this page needs an introductory component with the name of the category and how many materials are available under this category.
3. Third we need a search bar, no need to add submit button.
4. Fourth we need a component for sorting and filtering i.e. a sort bar, no need to add any functionalities just yet. This component, needs buttons for sorting by type, date, branch, year.
5. Fifth we need a component to view all the materials in the form of cards, the card needs to have a title, date of upload, a preview button and a download button as well as a bookmark button/icon.
6. Sixth is the footer component same as the one used in the landing page.

For reference - https://meganmartin.net/wp-content/uploads/2019/10/The-2-Types-of-Resources-Pages-and-Which-One-is-Right-for-Your-Website-6.jpg

Figma Link - https://www.figma.com/file/BFLdR6rSd6rqjsx3nYepUT/UniShare?type=design&node-id=0-1&t=Aubr0bmQO9JWssuZ-0

### PR DESCRIPTION
**Pull request by Akashsah2003**
## Overview

1. This PR fixes or fixes part of #13 .
2. This PR does the following: Created a materials listing page.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [x] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct

https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/2225eefe-c916-4d2a-84dc-97af7bbd9c5c

## PR Pointers

- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL".
- Never force push. If you do, your PR will be closed.

@Anurag607 @sambhavgupta0705 Please review this. Also was working on caching the fetched data but shouldn't there be an expiry time on the data as we would need to fetch fresh data after a time interval. Specify the time interval and any more feedbacks. After that I will add function for caching too.

### CODE REVIEW

**Comment by Anurag607** [comment]
**@Akashsah2003 please add a little horizontal padding to the cards.**
Aside from that i think everything else looks good.
@sambhavgupta0705 kindly take a look.

**Comment by Anurag607** [comment]
@Akashsah2003, since this site will see frequent use by the clients i think keeping a moderately long expiration duration is more suitable for example **1week**.

**Comment by Anurag607** [comment]
@Akashsah2003 let's keep the expiry duration at around **3days**. LMK your thoughts @sambhavgupta0705 .

**Comment by sambhavgupta0705** [comment]
@Akashsah2003 Kindly resolve conflicts and please share  a new preview

File: src/config/CardData.mjs
Comment (sambhavgupta0705) [inline src/config/CardData.mjs:1]: IMO we should use json format

Comment (Anurag607) [reply src/config/CardData.mjs:1]: I think it's alright, it's easier to traverse and manipulate arrays compared to objects.
I mean in the end we always end up converting objects to array (.entries()) 😅

Comment (sambhavgupta0705) [reply src/config/CardData.mjs:1]: Okayy @Anurag607

**Comment by sambhavgupta0705** [comment]
@Akashsah2003 kindly resolve merge conflicts

**Comment by Akashsah2003** [comment]
@Anurag607 @sambhavgupta0705 I have resolved the conflict and added caching as well as padding. 

But I think that caching data for 3 days can pose an issue for the user as he will be seeing 3 days old data whereas resources will be updated in a regular basis.
Kindly provide suggestions and reviews on this.

This is the preview.

https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/fa14d25c-99d1-48e6-a97e-c43b9c186a4d

**Comment by sambhavgupta0705** [comment]
@Akashsah2003 that's not how caching works if the database is updated then cached data is also updated.

**Comment by Akashsah2003** [comment]
> @Akashsah2003 that's not how caching works if the database is updated then cached data is also updated.

Ok got it. @sambhavgupta0705  @Anurag607 I am thinking about using react query to do the fetching and caching tasks. Does this approach seem good to you?

**Comment by sambhavgupta0705** [comment]
@Akashsah2003 @Anurag607 should we worry about caching for now.
First we are trying to have a layour of the site .
We first need to add firebase auth and cloudinary support ,all these cache and localstorage are secondary priority for us.

**Comment by Anurag607** [comment]
@sambhavgupta0705 you are quite right.
@Akashsah2003 let's skip the caching part for now. If there are any more updates regarding the UI please let us know, otherwise let's wrap it up as quickly as possible.

**Comment by Anurag607** [comment]
@Akashsah2003 kindly remove the caching part for now, we will take care of it later on once the backend is setup

**Comment by Akashsah2003** [comment]
@Anurag607 @sambhavgupta0705 I have removed caching. This is the latest preview.

https://github.com/bsoc-bitbyte/resource-sharing/assets/73382164/0f19cb75-00bc-40c5-a182-bf2fff0a6a4d

Please review and merge the pr.

**Review by sambhavgupta0705 — COMMENTED** [review]
Approved 👍🏻 
Thanks @Akashsah2003

**Review by sambhavgupta0705 — APPROVED** [review]
Approved

**Pull request merged**